transformer/events_creator_test.go

```diff
@@ -41,61 +48,113 @@ var fetcherResult = fetchers.FileSystemResource{
 }
 
 var (
-	opaResults   RuleResult
-	resourcesMap = map[string][]fetchers.FetchedResource{fetchers.FileSystemType: {fetcherResult}}
-	ctx          = context.Background()
-	events       = make([]beat.Event, 0)
-	cycleId      uuid.UUID
+	opaResults      evaluator.RuleResult
+	mockedEvaluator = evaluator.MockedEvaluator{}
+	resourcesMap    = map[string][]fetchers.FetchedResource{fetchers.FileSystemType: {fetcherResult}}
+	ctx             = context.Background()
 )
 
-func TestMain(m *testing.M) {
-	// Collect data before the test
-	cycleId, _ = uuid.NewV4()
+type EventsCreatorTestSuite struct {
+	suite.Suite
+	cycleId         uuid.UUID
+	mockedEvaluator evaluator.MockedEvaluator
+}
+
+func TestSuite(t *testing.T) {
+	suite.Run(t, new(EventsCreatorTestSuite))
+}
+
+func (s *EventsCreatorTestSuite) SetupSuite() {
 	parseJsonfile(opaResultsFileName, &opaResults)
+	s.cycleId, _ = uuid.NewV4()
+}
 
-	m.Run()
+func (s *EventsCreatorTestSuite) SetupTest() {
+	s.mockedEvaluator = evaluator.MockedEvaluator{}
 }
 
-func TestTransformer_ProcessAggregatedResources(t *testing.T) {
-	tests := []testAttr{
+func (s *EventsCreatorTestSuite) TestTransformer_ProcessAggregatedResources() {
+	var tests = []testAttr{
```

Comment:
I am less admirer of this practice although that's what Go recommended.

I like it better to have multiple tests with different names that describe what the test tries to test, but your code is totally legit.

I feel that here the test cases are not testing the same exact thing, so I would expect here to see three tests.

But again - it's just a personal flavor - your code is totally legit.

Reply:
I would have extracted it to another variable at least, just for the sake of readability - but again - my personal flavor.